internal/runners/activate/activate.go

```diff
@@ -81,12 +95,28 @@ func (r *Activate) run(params *ActivateParams, activatorLoop activationLoopFunc)
 
 	projectPath := filepath.Dir(projectToUse.Source().Path())
 
-	// Send google analytics event with label set to project namespace
-	names, fail := project.ParseNamespaceOrConfigfile(params.Namespace.String(), filepath.Join(projectPath, constants.ConfigFileName))
-	if fail != nil {
-		names = &project.Namespaced{}
-		logging.Debug("error resolving namespace: %v", fail.ToError())
+	names := params.Namespace
+	if params.Replace {
+		if names.CommitID == nil || *names.CommitID == "" {
+			latestID, fail := model.LatestCommitID(names.Owner, names.Project)
+			if fail != nil {
+				return locale.WrapInputError(fail.ToError(), "err_set_namespace_retrieve_commit", "Could not retrieve the latest commit for the specified project {{.V0}}.", names.String())
+			}
+			names.CommitID = latestID
+		}
+		err := projectToUse.Source().SetNamespaceAndCommit(names.Owner, names.Project, names.CommitID.String())
+		if err != nil {
+			return locale.WrapError(err, "err_activate_replace_write_commit", "Failed to write new namespace to activestate.yaml.")
+		}
+	} else {
+		var fail *failures.Failure
+		names, fail = project.ParseNamespaceOrConfigfile(params.Namespace.String(), filepath.Join(projectPath, constants.ConfigFileName))
+		if fail != nil {
+			names = &project.Namespaced{}
+			logging.Debug("error resolving namespace: %v", fail.ToError())
+		}
```

Comment:
This is way too much going on for this function. Please split it off into a separate function.